File: VB/CustomControlDesigner.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections
Imports System.Text
Imports System.Drawing
Imports DevExpress.XtraReports
Imports DevExpress.XtraReports.UI
Imports DevExpress.XtraReports.Design
Imports DevExpress.XtraReports.Serialization
Imports DevExpress.XtraReports.Native
Imports DevExpress.Utils.Design
Imports System.ComponentModel
Imports System.ComponentModel.Design
Imports RepCustomXRRichText.ControlsFamily
Imports DevExpress.XtraReports.UserDesigner

Namespace RepCustomXRRichText.Design
	''' <summary>
	''' Summary description for CustomControlDesigner.
	''' </summary>
	Public Class CustomControlDesigner
		Inherits XRRichTextDesigner
		Public Shared MyCustomEditor As Integer = &H77

		Public Sub New()

		End Sub

		Public Overrides Sub Initialize(ByVal component As IComponent)
			MyBase.Initialize(component)
			Me.Verbs.Add(New DesignerVerb("MyCustomEditor", New EventHandler(AddressOf OnMyCustomEditor), New CommandID(New Guid(), MyCustomEditor)))

			Dim i As Integer = 0
			Do While i < Me.Verbs.Count
				Dim verb As XRDesignerVerb = (TryCast(Me.Verbs(i), XRDesignerVerb))

				If verb IsNot Nothing AndAlso verb.ReportCommand = ReportCommand.VerbEditText Then
					Me.Verbs.Remove(verb)
					Exit Do
				End If
				i += 1
			Loop

		End Sub

		Public Sub OnMyCustomEditor(ByVal sender As Object, ByVal e As EventArgs)
			Dim frm As New RepCustomXRRichText.RichTextEditorForm()

			frm.richTextBox1.Rtf = (TryCast(Me.Component, XRRichText)).Rtf
			frm.Owner = (TryCast(Me.ReportDesigner.RootReport.Tag, System.Windows.Forms.Form))
			AddHandler frm.FormClosed, AddressOf frm_FormClosed
			frm.Show()
		End Sub

		Private Sub frm_FormClosed(ByVal sender As Object, ByVal e As System.Windows.Forms.FormClosedEventArgs)
			TryCast(Me.Component, XRRichText).Rtf = TryCast(sender, RepCustomXRRichText.RichTextEditorForm).richTextBox1.Rtf
			TryCast(Me.GetService(GetType(IBandViewInfoService)), IBandViewInfoService).UpdateView()
		End Sub

		Protected Overrides Sub RegisterActionLists(ByVal list As DesignerActionListCollection)
			MyBase.RegisterActionLists(list)
		End Sub
	End Class


End Namespace

File: CS/CustomControlDesigner.cs
using System;
using System.Collections;
using System.Text;
using System.Drawing;
using DevExpress.XtraReports;
using DevExpress.XtraReports.UI;
using DevExpress.XtraReports.Design;
using DevExpress.XtraReports.Serialization;
using DevExpress.XtraReports.Native;
using DevExpress.Utils.Design;
using System.ComponentModel;
using System.ComponentModel.Design;
using RepCustomXRRichText.ControlsFamily;
using DevExpress.XtraReports.UserDesigner;

namespace RepCustomXRRichText.Design {
	/// <summary>
	/// Summary description for CustomControlDesigner.
	/// </summary>
	public class CustomControlDesigner : XRRichTextDesigner {
        public static int MyCustomEditor = 0x77;
              
        public CustomControlDesigner() {
			
		}

        public override void Initialize(IComponent component) {
            base.Initialize(component);
            this.Verbs.Add(new DesignerVerb("MyCustomEditor", new EventHandler(OnMyCustomEditor), new CommandID(new Guid(), MyCustomEditor)));

            for (int i = 0; i < this.Verbs.Count; i++) {
                XRDesignerVerb verb = (this.Verbs[i] as XRDesignerVerb);

                if (verb != null && verb.ReportCommand == ReportCommand.VerbEditText) {
                    this.Verbs.Remove(verb);
                    break;
                }
            }

        }

        public void OnMyCustomEditor(Object sender, EventArgs e) {
            RepCustomXRRichText.RichTextEditorForm frm = new RepCustomXRRichText.RichTextEditorForm();

            frm.richTextBox1.Rtf = (this.Component as XRRichText).Rtf;
            frm.Owner = (this.ReportDesigner.RootReport.Tag as System.Windows.Forms.Form);
            frm.FormClosed += new System.Windows.Forms.FormClosedEventHandler(frm_FormClosed);
            frm.Show();
        }

        void frm_FormClosed(object sender, System.Windows.Forms.FormClosedEventArgs e) {
            (this.Component as XRRichText).Rtf = (sender as RepCustomXRRichText.RichTextEditorForm).richTextBox1.Rtf;
            (this.GetService(typeof(IBandViewInfoService)) as IBandViewInfoService).UpdateView();
        }

		protected override void RegisterActionLists(DesignerActionListCollection list) {
            base.RegisterActionLists(list);		
		}
	}

	
}
